doc/adrs/107_add_support_for_transient_data_in_forms.adoc

```diff
@@ -0,0 +1,84 @@
+= [ADR-107] Link widgets together in forms
+
+== Context
+
+The forms that can be defined in Sirius Web are sharing a core restriction with the form from EEF.
+In both case, you can only create forms using data backed up by models.
+As a result, there are families of use cases that are complex or borderline impossible to support.
+In some use cases, the end users may want to manipulate, in their forms, transient data which are not persisted in a model.
+
+Among those use case, we can find a dialog to create a new element.
+We don't want to create the element directly when we open the dialog and edit its properties.
+We want to create the element only when the user will have click on "Ok".
+
+Another popular use case is the creation of master / details interfaces.
+In this case, we could have a select showing a list of objects and below various widgets displaying the properties of the selected object.
+While the data in the various widgets would come from the model, the currently selected object should not be persisted in the model at all.
```

Comment:
> the currently selected object should not be persisted in the model at all

I understand, but the wording is ambibuous, because the object itself *is* persisted. What is not is the knowledge about which element is currently selected.